lib/segment/src/id_tracker/id_tracker_base.rs

```diff
@@ -111,8 +118,10 @@ pub trait IdTracker {
         &'a self,
         deleted_vector_bitslice: Option<&'a BitSlice>,
     ) -> Box<dyn Iterator<Item = PointOffsetType> + '_> {
+        // Use seeded randomness, prevents 'inconsistencies' in search results with sampling
+        let mut rng = StdRng::seed_from_u64(SEED);
+
         let total = self.total_point_count() as PointOffsetType;
-        let mut rng = rand::thread_rng();
         Box::new(
             (0..total)
                 .map(move |_| rng.gen_range(0..total))
```

Comment:
It's not related to the bug. instead of `rng.gen_range` that can generate the same id's I propose using this crate
https://docs.rs/permute/latest/permute/
It's used in quantization for taking subset of id's here:
https://github.com/qdrant/quantization/blob/master/quantization/src/quantile.rs#L31

Reply:
There's also this: https://docs.rs/rand/latest/rand/seq/index/fn.sample.html

Reply:
Good proposal, may integrate it in quantization lib. But anyway I guess it's a good time to change random selection

Reply:
@IvanPleshkov Instead I'm looking at `Permutor` now which is part of `permutation_iterator`:

https://docs.rs/permutation_iterator/latest/permutation_iterator/struct.Permutor.html

> There's also this: https://docs.rs/rand/latest/rand/seq/index/fn.sample.html

@ffuugoo We _never_ exhaust these samples items. In fact, we only ever use a few items from this sampled list. We don't know how much items to sample, because we don't know how many will be used at this stage. We could sample `total` number of items, but that would be very expensive :disappointed:

Thanks for the suggestion though!

Reply:
@timvisee `permutation_iterator` is 🔥🥵🤯

Reply:
I did some testing before committing to it. [`Permutor` ](https://docs.rs/permutation_iterator/latest/permutation_iterator/struct.Permutor.html) is cool stuff, but it is slow.

Here's a criterion benchmark on 100k items, reading 1k items, sampling the benchmark 1k times:

```
rand_test/seeded_rng    time:   [3.7673 µs 3.7704 µs 3.7733 µs]
rand_test/seeded_rng_sample
                        time:   [728.34 µs 728.47 µs 728.60 µs]
rand_test/permutor      time:   [8.1937 ms 8.1963 ms 8.1990 ms]
```

- Using `Permutor` is 2.1k times slower than using the naive `rng.gen_range(..)`.
- Using `sample` on 100k items 200 times slower than the naive `rng.gen_range(..)`.

Since this can be a hot path, I'll stick with `rng.gen_range(..)` and accept the potential collisions.

@IvanPleshkov @ffuugoo What do you think? Stick with `rng.gen_range(..)` as well?

Reply:
I'm sure that we have to keep it as is after benchmarks. Thanks for benches @timvisee! We don't have to pay so much performance for small possibility of better cardinality accuracy

Reply:
Sorry for wasting your time to this non-bug related discussion

Reply:
> Sorry for wasting your time to this non-bug related discussion

No problem.

It would have been an fantastic improvement. Too bad it is so expensive.

@IvanPleshkov @ffuugoo Thanks a lot for your suggestions.